lightningd/plugin_hook.c

```diff
@@ -193,18 +219,21 @@ void plugin_hook_db_sync(struct db *db)
 	struct jsonrpc_request *req;
 	struct plugin_hook_request *ph_req;
 	void *ret;
+	struct plugin *plugin;
 
 	const char **changes = db_changes(db);
-	if (!hook->plugin)
+	if (tal_count(hook->plugins) == 0)
 		return;
 
-	ph_req = notleak(tal(hook->plugin, struct plugin_hook_request));
+	ph_req = notleak(tal(hook->plugins, struct plugin_hook_request));
```

Comment:
This is a subtle change: logically it would become `tal(hook->plugins[0]`. If plugin was freed, the ph_req would go with it.

Reply:
... but I see how this works, never mind.